### ISSUE TITLE
Number validation not showing when also required

### ISSUE THREAD

**Issue by abobwhite** [jonsamwell/angular-auto-validate#55]
I just ran across an issue - and I'm not sure if it's an angular issue or angular-auto-validate. I have an ```<input type="number" required />``` and I seem to always be getting the required error validation message even i've typed something (non-numeric) into the input. I am expecting the error message to be the number error message, not the required message.

Thoughts?

**Comment by jonsamwell** [comment]
Hmmm, most likely if you put a non-numeric into the field the formatter in angular is failing so the model is not updating so it appears to auto-validate that the field has not got anything in.  I'll investigate.

**Comment by abobwhite** [comment]
Thanks. I found a workaround by creating a directive ```allow-invalid``` which always returns ```ngModelCtrl.$viewValue``` from ```$parsers.push``` that seems to put the invalid value on the model but it's kind of a hack. The only reason I believe this could be auto-validate issue is because it is behaving as expected on the above example in the angular documentation.

**Comment by jonsamwell** [comment]
Yeah this is a problem with the library specifically the debounce function on the ngModelDecorator.  Both the number and required validation errors are coming through but required is get picked over number.  Need to thing about handling this.

**Comment by abobwhite** [comment]
Any movement on this?

**Comment by jonsamwell** [comment]
Sorry I did look at this but haven't figure out a fix yet.

**Comment by nettino** [comment]
Any fix for this please?

**Comment by jonsamwell** [comment]
Sorry I completely forgot about this bug - I'll take a look
